### PR TITLE
[SVP-9436] Fix: Export as Email,HTML,PDF is ready for new api update + filters between tabs are memorized.

### PR DESCRIPTION
**Pull request by tlipczynski**

### CODE REVIEW

File: src/model/export-report/filterState.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export class FilterState {
```
Comment (kstepien-storware) [inline src/model/export-report/filterState.ts:1]: Implement `FilterStateInterface`

File: src/model/export-report/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  backupPolicyGuids: string[];
+  groupBy: string;
+  hypervisorClusterGuids: string[];
+  hypervisorGuids: string[];
+  hypervisorManagerGuids: string[];
+  projectGuids: string[];
+  virtualMachineGuids: string[];
+}
+
+export interface ExportRequestInterface {
```
Comment (kstepien-storware) [inline src/model/export-report/interfaces.ts:12]: you can put these interfaces into different files, then export them via `index.ts`

File: src/pages/dashboard/chargeback/SendReportViaEmailModal.tsx [hunk 3]
```diff
@@ -23,9 +25,23 @@ export default () => {
...
+      backupSize: mapPropertiesObjectListToStringOfGuids(
+        exportRequest.backupSize,
+      ),
+      transferSize: mapPropertiesObjectListToStringOfGuids(
+        exportRequest.transferSize,
+      ),
+      value: email.value,
+    };
+    if (exportRequestGuids.value.length > 0) {
+      await globalSettingsService.sendDashboardInfoEmail(
```
Comment (kstepien-storware) [inline src/pages/dashboard/chargeback/SendReportViaEmailModal.tsx:38]: `await globalSettingsService.sendDashboardInfoEmail(range, exportRequestGuids)`

File: src/pages/reporting/ExportAs.tsx [hunk 4]
```diff
@@ -30,20 +31,42 @@ export const ExportAs = () => {
...
+      backupSize: mapPropertiesObjectListToStringOfGuids(
+        exportRequest.backupSize,
+      ),
+      transferSize: mapPropertiesObjectListToStringOfGuids(
+        exportRequest.transferSize,
+      ),
+      value: '',
+    };
+    await fileSaverService.saveFile(
+      await dashboardService.getDashboardInfoPdf(
```
Comment (kstepien-storware) [inline src/pages/reporting/ExportAs.tsx:44]: same

File: src/pages/reporting/ExportAs.tsx [hunk 5]
```diff
@@ -30,20 +31,42 @@ export const ExportAs = () => {
...
+      backupSize: mapPropertiesObjectListToStringOfGuids(
+        exportRequest.backupSize,
+      ),
+      transferSize: mapPropertiesObjectListToStringOfGuids(
+        exportRequest.transferSize,
+      ),
+      value: '',
+    };
+    await fileSaverService.saveFile(
+      await dashboardService.getDashboardInfoHtml(
```
Comment (kstepien-storware) [inline src/pages/reporting/ExportAs.tsx:64]: same.

Put `dashboardService.getDashboardInfoHtml` into constant, then provide it into `saveFile`

File: src/pages/reporting/ExportAs.tsx [hunk 6]
```diff
@@ -58,12 +81,14 @@ export const ExportAs = () => {
...
 
   return (
-    <Dropdown
-      value={selectedExportType}
-      options={exportTypes}
-      optionLabel="description"
-      onChange={onExportTypeChange}
-      placeholder="Export"
-    />
+    <>
```
Comment (kstepien-storware) [inline src/pages/reporting/ExportAs.tsx:84]: you don't need this

File: src/pages/reporting/Reporting.tsx [hunk 7]
```diff
@@ -13,7 +13,7 @@ import {
   TransferSize,
 } from 'pages/reporting/Tabs';
 
-const tabs: Tab[] = [
+export const tabs: Tab[] = [
```
Comment (kstepien-storware) [inline src/pages/reporting/Reporting.tsx:16]: why export?

File: src/pages/reporting/components/ReportSizeContainer.tsx [hunk 8]
```diff
@@ -134,7 +164,9 @@ export const ReportSizeContainer = ({ chartData, getChargebackData }) => {
             <h3 className="mt-3">Filter</h3>
 
             {Object.keys(filterByFieldOptions).map((el) => {
```
Comment (kstepien-storware) [inline src/pages/reporting/components/ReportSizeContainer.tsx:166]: Rename "el" => "key"

**Review by kstepien-storware — CHANGES_REQUESTED** [review]

**Review by kstepien-storware — APPROVED** [review]

**Review by sebastsliwin97 — APPROVED** [review]

**Pull request merged**